fix(test): remove warning for side effect null in react dom

cbernat opened the pull request:
Some of the lengthy console warning that we are getting in tests log, are from third party libraries that we use, that haven't updated ther dependencies, this is why yarn allows a configuration in `package.json` we could update dependencies for a particular library. This obviously is temporal. https://yarnpkg.com/lang/en/docs/selective-version-resolutions/  

In tests for: shopify.test ,signup.test, Reports.test, App.test
```
● Console

    console.warn node_modules/react-dom/cjs/react-dom.development.js:12338
      Warning: componentWillMount has been renamed, and is not recommended for use. See https://fb.me/react-unsafe-component-lifecycles for details.

      * Move code with side effects to componentDidMount, and set initial state in the constructor.
      * Rename componentWillMount to UNSAFE_componentWillMount to suppress this warning in non-strict mode. In React 17.x, only the UNSAFE_ name will work. To rename all deprecated lifecycles to their new names, you can run `npx react-codemod rename-unsafe-lifecycles` in your project source folder.

      Please update the following components: SideEffect(NullComponent)
```

Pending 

- [x] add ^to version in resolution.
- [x] Check what we use in helmet is still funcional (title in pages, meta with description, zoho chat bubble logic, lang in html)

andresmoschini commented on yarn.lock:
```diff
@@ -11536,13 +11531,10 @@ react-scripts@^3.2.0:
-    exenv "^1.2.1"
-    shallowequal "^1.0.1"
+react-side-effect@2.1.0, react-side-effect@^1.1.0:
+  version "2.1.0"
```
If I am understanding it fine, it says that we will use the version `2.1.0` even when the client asks for version `1.1.0`.

It could have breaking changes (major version number has changed), is it acceptable? do you know if it is only being used in tests?

cbernat replied:
I will check, this was supposed to be only for react-helmet, not for all other packages.

cbernat replied:
It's only used by `react-helmet` which we use to update html document properties like title and meta tags.

andresmoschini replied:
The breaking change is related to the React version used, so, we are fine:

![image](https://user-images.githubusercontent.com/1157864/68386643-3c9f6a00-013b-11ea-890c-a2581781e19d.png)

fcoronelmakingsense commented on package.json:
```diff
@@ -108,5 +108,8 @@
     ]
+  },
+  "resolutions": {
+    "react-helmet/react-side-effect": "2.1.0"
```
it's only in this library the problem?

cbernat replied:
yes, I've searched through and found only helmet using it.

cbernat commented:
test this please.

andresmoschini commented on package.json:
```diff
@@ -108,5 +108,8 @@
     ]
+  },
+  "resolutions": {
+    "react-helmet/react-side-effect": "^2.1.0"
```
No le ponés un rango de versiones?

cbernat replied:
Pero funciona eso? te entendi que me pasaste algo en beta O_o

andresmoschini replied:
```suggestion
    "react-helmet/react-side-effect@>=1.0.0 <2.1.0": "react-helmet/react-side-effect@^2.1.0"
```

cbernat replied:
I've tried this and I get a warning:
```
yarn install v1.17.3
warning Resolution field "react-helmet/react-side-effect@^2.1.0" has an invalid version entry and may be ignored
``` 
So I'll leave this as is.

andresmoschini replied:
Did you try this?

```suggestion
    "react-helmet/react-side-effect@>=1.0.0 <2.1.0": "^2.1.0"
```

andresmoschini replied:
Si no hacemos algo así, cuando `react-helmet/react-side-effect` se actualice a la versión 3, vamos a seguir con la `2.x.x`

andresmoschini replied:
La otra opción es documentarnos mejor antes de hacer el cambio.

cbernat replied:
Esto ultimo si que funciona, gracias <3

cbernat replied:
Corri el test, recheckie el yarn lock y al final no funciona, sigo investigando. (en lo ultimo que subi se ve, en el lock)

cbernat commented:
test this please.

cbernat commented:
Test this please.

cbernat commented:
@andresmoschini I've updated the resolution to allow taking more than the actual version, also I've added a comment in the form of a property. What do you think?

andresmoschini commented:
> @andresmoschini I've updated the resolution to allow taking more than the actual version, also I've added a comment in the form of a property. What do you think?

I Agree. If we cannot find a better solution after spend some time on this, I think that it is an acceptable solution.

andresmoschini commented:
:tada: This PR is included in version 1.59.1 :tada:

The release is available on [GitHub release](https://github.com/FromDoppler/doppler-webapp/releases/tag/v1.59.1)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
